third_party/blink/renderer/modules/ml/v2/nn_model.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef THIRD_PARTY_BLINK_RENDERER_MODULES_ML_NN_MODEL_H_
+#define THIRD_PARTY_BLINK_RENDERER_MODULES_ML_NN_MODEL_H_
+
+#include "services/ml/public/mojom/model.mojom-blink.h"
+#include "third_party/blink/renderer/bindings/core/v8/script_promise.h"
+#include "third_party/blink/renderer/bindings/core/v8/script_promise_resolver.h"
+#include "third_party/blink/renderer/core/typed_arrays/dom_array_buffer_view.h"
+#include "third_party/blink/renderer/modules/ml/v2/compilation_options.h"
+#include "third_party/blink/renderer/modules/ml/v2/nn_context.h"
+#include "third_party/blink/renderer/platform/bindings/script_wrappable.h"
+#include "third_party/blink/renderer/platform/heap/heap_allocator.h"
+#include "third_party/blink/renderer/platform/heap/member.h"
+#include "third_party/blink/renderer/platform/wtf/vector.h"
+
+namespace blink {
+
+int32_t StringToOperandType(const String& operand_type);
+using NamedOperandVector = HeapVector<Member<NamedOperand>>;
+
+class NNModel final : public ScriptWrappable {
+  DEFINE_WRAPPERTYPEINFO();
+
+ public:
+  NNModel(ml::mojom::blink::ModelPtrInfo, NamedOperandVector*);
+  ~NNModel() override;
+
+  ScriptPromise createCompilation(ScriptState*,
+                                  const CompilationOptions* options);
+  void BuildNeuralNetworkModel(Operand*, ml::mojom::blink::ModelInfoPtr&);
+  void NameMapIndex(String name, uint32_t index);
+  void AddFuseOperand(ml::mojom::blink::ModelInfoPtr& model_info);
+  void AddUnspecifiedOperand(ml::mojom::blink::ModelInfoPtr& model_info);
+  void AddOperand(const OperandDescriptor* descriptor,
+                  ml::mojom::blink::ModelInfoPtr& model_info);
+  void SetOperandValue(uint32_t index,
+                       DOMArrayBufferView* data,
+                       ml::mojom::blink::ModelInfoPtr& model_info);
+  void AddOperation(int32_t type,
+                    const Vector<uint32_t>& inputs,
+                    const Vector<uint32_t>& outputs,
+                    ml::mojom::blink::ModelInfoPtr& model_info);
+  void IdentifyInputsAndOutputs(const Vector<uint32_t>& inputs,
+                                const Vector<uint32_t>& outputs,
+                                ml::mojom::blink::ModelInfoPtr& model_info);
```

Comment:
Could the `model_info` be internal to `NNModel`? So we don't need to pass it around in above methods.

Reply:
That's more clearly, thanks, done.